libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -222,7 +202,7 @@ TypePointer ReferencesResolver::typeFor(TypeName const& _typeName)
 
 			auto const* lengthType = dynamic_cast<IntegerConstantType const*>(length->annotation().type.get());
 			if (!lengthType)
-				BOOST_THROW_EXCEPTION(length->createTypeError("Invalid array length."));
+				fatalTypeError(length->location(), "Invalid array length.");
```

Comment:
Call to `location()` can be moved into `::typeError(...)`, but you can also leave it like that if you want.